Added gitreload state

bdero opened the pull request:
@blarghmatey This PR adds a state for installing gitreload, which is intended to be run on edX instances.

blarghmatey commented on salt/edx/gitreload.sls:
```diff
@@ -0,0 +1,84 @@
+
+create_gitreload_config:
+  file.managed:
+    - name: {{ salt.pillar.get('edx:gitreload:gr_dir') }}/gr.env.json
```
If you can specify the default value as the second parameter to the `pillar.get` function that would be useful. That way the pillar is just for overriding the defaults.

bdero commented:
I addressed the comment, moved the pillar data grabs to the top of the file, and added a comment to the slight known_hosts hack to explain the empty `context`.

blarghmatey commented:
👍
